sql/numbertype.go

```diff
@@ -86,6 +86,70 @@ func MustCreateNumberType(baseType query.Type) NumberType {
 	return nt
 }
 
+func NumericZeroValue(t Type) interface{} {
```

Comment:
Any reason you can't use Type.Zero()? This is just a subset of that, for the numeric types.